### PR TITLE
ready solution hw-5.2

### PR DESCRIPTION
**Pull request by RomanHreits**

### CODE REVIEW

File: src/test/java/core/basesyntax/PlaneTest.java [hunk 1]
```diff
@@ -63,9 +63,9 @@ public void checkThatAllFieldsFromPlaneExistsInBuilderAsSetters() {
         List<Class> planeInnerClasses = Arrays.asList(Plane.class.getClasses());
         Optional<Class> planeBuilderClass = planeInnerClasses.stream()
                 .filter(Objects::nonNull)
-                .filter(c -> c.getSimpleName().equals("PlaneBuilder"))
+                .filter(c -> c.getSimpleName().equals("Builder"))
                 .findFirst();
-        Assert.assertTrue("PlaneBuilder class should be present", planeBuilderClass.isPresent());
+        Assert.assertTrue("Builder class should be present", planeBuilderClass.isPresent());
```
Comment (boroda4436) [inline src/test/java/core/basesyntax/PlaneTest.java]: Please do not modify the test classes!

File: src/main/java/core/basesyntax/Plane.java [hunk 2]
```diff
@@ -1,11 +1,83 @@
...
  */
 
 public class Plane {
+    private String planeType;
+    private int numberOfPassengers;
+    private String yearOfProduction;
+    private String whichAirlines;
+    private int code;
 
+    private Plane(Builder builder) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Plane.java]: @RomanHreits please add the Main class with an example of how to use your builder

File: src/main/java/core/basesyntax/Plane.java [hunk 3]
```diff
@@ -1,11 +1,83 @@
 package core.basesyntax;
 
 /**
- * <p>Дано класс Plane. Написать класс PlaneBuilder, который реализует паттерн Builder.
+ * <p>Дано класс Plane. Написать класс Builder, который реализует паттерн Builder.
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Plane.java]: Please use refactoring carefully

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Plane.java (hunk 3 above)
Comment (RomanHreits) [reply src/main/java/core/basesyntax/Plane.java]: done

File: src/main/java/core/basesyntax/Plane.java (hunk 2 above)
Comment (RomanHreits) [reply src/main/java/core/basesyntax/Plane.java]: done

File: src/test/java/core/basesyntax/PlaneTest.java (hunk 1 above)
Comment (RomanHreits) [reply src/test/java/core/basesyntax/PlaneTest.java]: done

File: README.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# jv-builder
+# jv-planeBuilder
```
Comment (boroda4436) [inline README.md]: @RomanHreits )))

File: pom.xml [hunk 5]
```diff
@@ -5,7 +5,7 @@
     <modelVersion>4.0.0</modelVersion>
 
     <groupId>mate-academy</groupId>
-    <artifactId>builder</artifactId>
+    <artifactId>planeBuilder</artifactId>
```
Comment (boroda4436) [inline pom.xml]: @RomanHreits )))

File: src/main/java/core/basesyntax/Plane.java [hunk 6]
```diff
@@ -1,11 +1,83 @@
...
  */
 
 public class Plane {
+    private String planeType;
+    private int numberOfPassengers;
+    private String yearOfProduction;
+    private String whichAirlines;
+    private int code;
 
+    private Plane(PlaneBuilder planeBuilder) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Plane.java:16]: this makes no sense

File: src/main/java/core/basesyntax/Plane.java [hunk 7]
```diff
@@ -1,11 +1,83 @@
...
+                + "planeType= " + planeType + ", numberOfPassengers= " + numberOfPassengers
+                + ", yearOfProduction= " + yearOfProduction + ", whichAirlines= " + whichAirlines
+                + ", code=" + code + "}";
+    }
+
+    public static class PlaneBuilder {
+        private Plane plane;
+
+        public PlaneBuilder() {
+            plane = new Plane(this);
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Plane.java]: interesting solution

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (RomanHreits) [reply README.md]: done

File: pom.xml (hunk 5 above)
Comment (RomanHreits) [reply pom.xml]: done

File: src/test/java/core/basesyntax/PlaneTest.java [hunk 8]
```diff
@@ -41,7 +41,7 @@ public void checkThatBuilderClassIsPresent() {
         try {
             planeBuilderClass.get().getMethod("build");
         } catch (NoSuchMethodException e) {
-            Assert.fail("Builder class should contain method 'build'");
+            Assert.fail("PlaneBuilder class should contain method 'build'");
```
Comment (dmytro-hryhoruk) [inline src/test/java/core/basesyntax/PlaneTest.java]: don't modify test class

File: src/main/java/core/basesyntax/Plane.java [hunk 9]
```diff
@@ -1,11 +1,83 @@
...
 
 public class Plane {
+    private String planeType;
+    private int numberOfPassengers;
+    private String yearOfProduction;
+    private String whichAirlines;
+    private int code;
 
+    private Plane(PlaneBuilder planeBuilder) {
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Plane.java:22]: your solution works but this constructor is useless.

Comment (RomanHreits) [reply src/main/java/core/basesyntax/Plane.java:22]: done

File: src/test/java/core/basesyntax/PlaneTest.java (hunk 8 above)
Comment (RomanHreits) [reply src/test/java/core/basesyntax/PlaneTest.java]: okey

Comment (dmytro-hryhoruk) [reply src/test/java/core/basesyntax/PlaneTest.java]: return this file to the original state and make a commit. It should be gone from this PR

Comment (RomanHreits) [reply src/test/java/core/basesyntax/PlaneTest.java]: done

File: src/test/java/core/basesyntax/PlaneTest.java [hunk 10]
```diff
@@ -111,4 +111,4 @@ public void checkTheOnlyOnePlaneConstructorIsPresent() {
         Assert.assertEquals("Your constructor should have one parameter of PlaneBuilder class",
                 expectedPlaneBuilderClass, actualConstructorParameterType);
     }
-}
+}
```
Comment (boroda4436) [inline src/test/java/core/basesyntax/PlaneTest.java:114]: Please do not modify test class

**Review by boroda4436 — APPROVED** [review]

**Pull request closed without merging**